packages/shared-ux/page/solution_nav/src/with_solution_nav.tsx

```diff
@@ -8,24 +8,25 @@
 
 import React, { ComponentType, useState } from 'react';
 import classNames from 'classnames';
-import { useIsWithinBreakpoints } from '@elastic/eui';
+import { useIsWithinBreakpoints, EuiPageTemplateProps } from '@elastic/eui';
 import { EuiPageSideBarProps } from '@elastic/eui/src/components/page/page_side_bar';
-import { KibanaPageTemplateSolutionNav, KibanaPageTemplateSolutionNavProps } from './solution_nav';
-import { KibanaPageTemplateProps } from './types';
+import { SolutionNav, SolutionNavProps } from './solution_nav';
 
 // https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging
 function getDisplayName(Component: ComponentType<any>) {
   return Component.displayName || Component.name || 'UnnamedComponent';
 }
 
-type SolutionNavProps = KibanaPageTemplateProps & {
-  solutionNav: KibanaPageTemplateSolutionNavProps;
-};
+type Props<T> = T &
+  EuiPageTemplateProps & {
+    solutionNav: SolutionNavProps;
+    isEmptyState?: boolean;
+  };
 
 const SOLUTION_NAV_COLLAPSED_KEY = 'solutionNavIsCollapsed';
 
-export const withSolutionNav = (WrappedComponent: ComponentType<KibanaPageTemplateProps>) => {
-  const WithSolutionNav = (props: SolutionNavProps) => {
+export const withSolutionNav = <T,>(WrappedComponent: ComponentType) => {
```

Comment:
@chandlerprall Is there an inference I can use here, rather than explicit generic?  I was getting that weird instantiation error again.